### DISCUSSION TITLE
when -resample , the print size changed ! How to avoid？

### DISCUSSION THREAD

**Discussion by mike-kong** [ImageMagick/ImageMagick#6238] [Help]
### ImageMagick version

7.1.0-61

### Operating system, version and so on

windows

### Description

![image](https://user-images.githubusercontent.com/92786267/230534520-b99b8b96-96de-4a2a-95b9-625050aa6b8d.png)
**

> identify -verbose  "C:\Users\86131\Desktop\新建文件夹\222 - 副本.png"

**
identify -verbose get a wrong resolution
{
          Geometry: 804x804+0+0
          Resolution: **96x96**  
          Print size: **8.375x8.375**
          Units: **PixelsPerInch**
}
![image](https://user-images.githubusercontent.com/92786267/230534585-378663aa-de77-467f-93dd-ccddea880697.png)
 identify -verbose -ping get the right resolution

>  identify -verbose -ping "C:\Users\86131\Desktop\新建文件夹\222 - 副本.png"

{
          Geometry: 804x804+0+0
          Resolution: **38.97x38.97**
          Print size: **20.6313x20.6313**
          Units: **PixelsPerCentimeter**
 }
 **right size is 804px , 38.97px/cm=99px/inch , Print size:20.6313cm=8.123inch**

![222 - 副本](https://user-images.githubusercontent.com/92786267/230534651-905d65bf-2a4d-483e-88f0-54b8864ecd8c.png)

> convert "C:\Users\86131\Desktop\新建文件夹\222 - 副本.png"    -resample 300 "C:\Users\86131\Desktop\新建文件夹\222kkkk - 副本.png"

I want to resample it , The result I expect is 
{
        Geometry: 2436x2436+0+0
        Resolution: 118.09x118.09
        Print size: 20.6313x20.6313
        Units: PixelsPerCentimeter
}
but the result trun out :
{
 Geometry: 2513x2513+0+0
  Resolution: 118.11x118.11
  Print size: 21.2768x21.2768
  Units: PixelsPerCentimeter
}

**### I guss the reason is ,  when resample , it based on the wrong Resolution 96px/inch=37.80px/inch
My question is how can i tell the system , to based on the right Resolution 99px/inch=38.97px/inch**

**Comment by snibgo** [comment]
If you want to change just the resolution (eg PixelsPerInch) without also changing the number of pixels, use "-density 99 -units PixelsPerInch" or whatever you need.

**Reply by mike-kong** [reply]
thankyou for your reply . 
sorry ,  I want to keep the print size in cm/inch 
change the pixels
that's why I use resample 
 but the problem is when I use resample , the  print size in cm/inch  changed . 
I guess the reason is system based on the wrong resolution when conducting resample 
but I don't know how tell the system the right resolution so that it can  conducting resample  base on the right resolution .

**Comment by fmw42** [comment]
What was your exact command line?  If you save to PNG from some other format, you might be seeing a change in resolution since PNG does not support pixelsperinch. It is automatically converted to pixelspercentimeterd.
